src/game/win_detection_test.go

```diff
@@ -0,0 +1,102 @@
+package game
+
+import (
+    "testing"
+	"fmt"
+)
+
+var detectWinnerDatasets = []struct {
+	in  string
+	out string
+}{
+	{`
+	┌────────────┐
+	|1|1|1||1|1|0|
+	|0|0|0||0|0|0|
+	|0|0|0||0|0|0|
+	|────────────|
+	|0|0|0||0|0|0|
+	|0|0|0||0|0|0|
+	|0|0|0||0|0|0|
+	└────────────┘`, "GAME_PLAYER1_WON"},
+	{`
+	┌────────────┐
+	|1|0|1||1|1|0|
+	|0|1|0||0|0|0|
+	|0|0|1||0|0|0|
+	|────────────|
+	|0|0|0||1|0|0|
+	|0|0|0||0|1|0|
+	|0|0|0||0|0|0|
+	└────────────┘`, "GAME_PLAYER1_WON"},
+	{`
+	┌────────────┐
+	|1|1|1||1|1|0|
+	|0|0|0||0|0|0|
+	|0|0|0||0|0|0|
+	|────────────|
+	|0|0|0||0|0|0|
+	|0|0|0||0|0|0|
+	|2|2|2||2|2|0|
+	└────────────┘`, "GAME_DRAW"},
+	{`
+	┌────────────┐
+	|0|0|0||0|0|0|
+	|0|2|0||0|0|0|
+	|0|2|0||0|0|0|
+	|────────────|
+	|0|2|0||0|0|0|
+	|0|2|0||0|0|0|
+	|1|2|0||1|1|0|
+	└────────────┘`, "GAME_PLAYER2_WON"},
+	{`
+	┌────────────┐
+	|0|0|0||0|0|0|
+	|0|0|0||0|1|0|
+	|0|1|0||1|0|0|
+	|────────────|
+	|0|1|1||0|0|0|
+	|0|1|0||0|0|0|
+	|1|1|0||1|1|0|
+	└────────────┘`, "GAME_PLAYER1_WON"},
+	{`
+	┌────────────┐
+	|0|2|2||1|2|2|
+	|1|1|2||2|1|1|
+	|1|1|2||1|1|1|
+	|────────────|
+	|2|1|1||2|2|2|
+	|1|2|2||1|2|2|
+	|1|1|1||1|2|2|
+	└────────────┘`, "GAME_RUNNING"},
+	{`
+	┌────────────┐
+	|1|2|2||1|2|2|
+	|1|1|2||2|1|1|
+	|1|1|2||1|1|1|
+	|────────────|
+	|2|1|1||2|2|2|
+	|1|2|2||1|2|2|
+	|1|1|1||1|2|2|
+	└────────────┘`, "GAME_DRAW"},
+}
+
+func TestDetectWinner(t *testing.T) {
```

Comment:
I prefer to have multiple explicit and simple tests rather than a complex for loop. It's easier to identify what is wrong.